### ISSUE TITLE
Ovpn Feature

### ISSUE THREAD

**Issue by 6abd** [6abd/horus#10]
Connect to a specified openvpn

**Comment by tpegl** [comment]
Hi, I was hoping to be able to pick this up but I made a start before commenting on here which is bad practice, so apologies for that.

Was the idea for this to be something similar to how the Proton VPN implementation is done or something more involved (i.e. setting up for a specific provider or setting up the connection to a custom host rather than just using an OpenVPN profile)? I couldn't find any particularly solid, up-to-date packages for OpenVPN besides some management ones rather than purely for connecting.

Thanks for the cool project!

**Comment by 6abd** [comment]
Hey, thanks for commenting and taking a look at the project.

Yes, the idea was for it to be a similar implementation as ProtonVPN was done. However, it was just an idea I thought could be done, as are a handful of the features I haven't implemented yet. If there are no well-documented libraries for it, an idea I had was to just combine VPN features like you mentioned into one command (e.g. adding subfeatures like connecting to different providers). They could also just be in different commands too. 

To summarize, the original idea was for Ovpn to be a similar implementation as Pvpn, however without a well-documented library you can try other providers that do have libraries, and/or implementing other VPN features.

If you want to contribute you can work on that or any other feature ideas. You can also come up with your own feature if you have any ideas. Any contributions are appreciated, and thanks for showing interest!

**Comment by tpegl** [comment]
That all makes sense, I was just going to call standard `openvpn` CLI from `subprocess.Popen`. Was the idea also that someone would call that, have a VPN connection and then do other things with the library so you'd want the process to be run in the background?

I'm really happy I found this repo, I'm a fan of OSINT and just starting out actually contributing to open-source so this is perfect! More than happy to pick up other features as well, I was thinking OnionShare next since that sounds very interesting.

**Comment by 6abd** [comment]
Yeah, the idea would be for it to be run in the background, though I guess users could use multiple terminal windows.

I'm glad you like the repo and find it interesting! You can work on any features you want to/are interested in. It would actually help a lot as I've been busy with school. Don't feel any pressure with this though, and feel free to ask me for help. I'm still learning too, so if you find anything that needs to be changed just let me know. My contact info is on the ReadME.
